clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractFetch.java

```diff
@@ -97,31 +81,13 @@ public AbstractFetch(final LogContext logContext,
         this.subscriptions = subscriptions;
         this.fetchConfig = fetchConfig;
         this.decompressionBufferSupplier = BufferSupplier.create();
-        this.completedFetches = new ConcurrentLinkedQueue<>();
+        this.fetchBuffer = new FetchBuffer<>(logContext);
         this.sessionHandlers = new HashMap<>();
         this.nodesWithPendingFetchRequests = new HashSet<>();
         this.metricsManager = metricsManager;
         this.time = time;
     }
 
-    /**
-     * Return whether we have any completed fetches pending return to the user. This method is thread-safe. Has
-     * visibility for testing.
-     *
-     * @return true if there are completed fetches, false otherwise
-     */
-    boolean hasCompletedFetches() {
```

Comment:
Just wondering, is there a reason to move these 2 to the child class Fetcher? I see they just perform some action on what is now the FetcherBuffer, that is defined here in the AbstractFetch, so I would say it still makes sense to have the 2 functions here, and avoid an unneeded change maybe?

Reply:
Thanks for catching that. Let me try to remember why I moved them...

Reply:
I have carefully set these two methods back where I found them 😄